### PR TITLE
fix(api-reference): initial dark mode state bug and add configuration for darkmode

### PR DESCRIPTION
**Pull request by marclave**
**Problem**
Currently users arent able to disable the dark mode button or force a state, this often leads to us defaulting to the browser or system settings which can be infinitely jarring

**Solution**
I updated the docs, added some more configuration for the user :) we are so back

#### darkMode?: boolean

Whether dark mode is on or off initially (light mode)

```js
{
  darkMode: true
}
```

#### forceDarkModeState?: 'dark' | 'light'

forceDarkModeState makes it always this state no matter what

```js
{
  forceDarkModeState: 'dark'
}
```

#### hideDarkModeToggle?: boolean

Whether to show the dark mode toggle

```js
{
  hideDarkModeToggle: true
}
```

some people might have external night and day mode buttons, so these arent all inclusive :)

### CODE REVIEW

File: packages/api-reference/src/hooks/useDarkModeState.ts [hunk 1]
```diff
@@ -46,13 +49,17 @@ export function useDarkModeState(isDarkInitially?: boolean) {
...
-  // Priority of initial values is: LocalStorage/App Config/Fallback
-  isDark.value =
-    (typeof window === 'undefined'
-      ? null
-      : JSON.parse(window.localStorage?.getItem('isDark') || 'null')) ??
-    isDarkInitially ??
-    getDarkModeState()
+  // Priority of initial values is: forceState -> LocalStorage/App ->  Config/Fallback
+  if (forceDarkModeState) {
+    isDark.value = forceDarkModeState === 'dark' ? true : false
```
Comment (amritk) [inline packages/api-reference/src/hooks/useDarkModeState.ts:54]:
```suggestion
    isDark.value = forceDarkModeState === 'dark' 
```

File: packages/api-reference/src/components/Layouts/ClassicLayout.vue [hunk 2]
```diff
@@ -41,6 +41,7 @@ const config = computed(() => ({ ...props.configuration, showSidebar: false }))
           :spec="spec" />
         <template #dark-mode-toggle>
           <DarkModeIconToggle
+            v-if="!!!props.configuration.hideDarkModeToggle"
```
Comment (amritk) [inline packages/api-reference/src/components/Layouts/ClassicLayout.vue:44]:
```suggestion
            v-if="!props.configuration.hideDarkModeToggle"
```

File: packages/api-reference/src/components/Layouts/ModernLayout.vue [hunk 3]
```diff
@@ -62,6 +62,7 @@ watch(hash, (newHash, oldHash) => {
     </template>
     <template #sidebar-end>
       <DarkModeToggle
+        v-if="!!!props.configuration.hideDarkModeToggle"
```
Comment (amritk) [inline packages/api-reference/src/components/Layouts/ModernLayout.vue:65]:
```suggestion
        v-if="!props.configuration.hideDarkModeToggle"
```

File: packages/api-reference/src/hooks/useDarkModeState.ts [hunk 4]
```diff
@@ -46,13 +49,17 @@ export function useDarkModeState(isDarkInitially?: boolean) {
...
+  } else {
+    isDark.value =
+      (typeof window === 'undefined'
+        ? null
+        : JSON.parse(window.localStorage?.getItem('isDark') || 'null')) ??
+      isDarkInitially ??
+      getDarkModeState()
+  }
```
Comment (amritk) [inline packages/api-reference/src/hooks/useDarkModeState.ts:62]: this bit looks a bit messy and is hard to follow, maybe re-write?

**Review by amritk — CHANGES_REQUESTED** [review]
some minor code style changes

File: packages/api-reference/src/hooks/useDarkModeState.ts (hunk 4 above)
Comment (marclave) [reply packages/api-reference/src/hooks/useDarkModeState.ts:62]: this existed before the PR

File: documentation/configuration.md [hunk 5]
```diff
@@ -90,6 +90,36 @@ Whether to show the "Download OpenAPI Specification" button
 }
 ```
 
+#### darkMode?: boolean
+
+Whether dark mode is on or off initially (light mode)
```
Comment (hanspagel) [inline documentation/configuration.md:95]: What does the `(light mode)` mean? This confuses me. Maybe something like that:

> Initially set dark mode (`true`) or light mode (`false`).

File: documentation/configuration.md [hunk 6]
```diff
@@ -90,6 +90,36 @@ Whether to show the "Download OpenAPI Specification" button
...
+
+```js
+{
+  darkMode: true
+}
+```
+
+#### forceDarkModeState?: 'dark' | 'light'
+
+forceDarkModeState makes it always this state no matter what
```
Comment (hanspagel) [inline documentation/configuration.md:105]:
```suggestion
`forceDarkModeState` makes it always `'dark'` or `'light'` mode, no matter what.
```

File: documentation/configuration.md (hunk 5 above)
Comment (hanspagel) [inline documentation/configuration.md:95]: What does the `(light mode)` mean? This confuses me. Maybe something like that:

```suggestion
Initially set dark mode (`true`) or light mode (`false`).
```

File: documentation/configuration.md [hunk 7]
```diff
@@ -90,6 +90,36 @@ Whether to show the "Download OpenAPI Specification" button
...
+
+```js
+{
+  forceDarkModeState: 'dark'
+}
+```
+
+#### hideDarkModeToggle?: boolean
+
+Whether to show the dark mode toggle
```
Comment (hanspagel) [inline documentation/configuration.md:115]:
```suggestion
By default, a toggle to switch between light and dark mode is shown. Passing `true` hides this toggle.
```

**Review by amritk — COMMENTED** [review]
Remove requsted changes
